File: backend/internal/models/activity.go
package models

// ActivityItem is one row for recent on-chain activity (signature-level).
type ActivityItem struct {
	Signature            string `json:"signature"`
	Slot                 uint64 `json:"slot"`
	BlockTime            *int64 `json:"block_time,omitempty"`
	Success              bool   `json:"success"`
	ErrorSummary         string `json:"error_summary,omitempty"`
	Memo                 string `json:"memo,omitempty"`
	SolscanURL           string `json:"solscan_url"`
	ConfirmationStatus string `json:"confirmation_status,omitempty"`
}

// ActivityResponse is returned by GET /api/v1/wallet/:address/activity
type ActivityResponse struct {
	WalletAddress string         `json:"wallet_address"`
	Items         []ActivityItem `json:"items"`
	NextBefore    string         `json:"next_before,omitempty"`
}

File: frontend/src/types/activity.ts
export interface ActivityItem {
  signature: string;
  slot: number;
  block_time?: number;
  success: boolean;
  error_summary?: string;
  memo?: string;
  solscan_url: string;
  confirmation_status?: string;
}

export interface ActivityResponse {
  wallet_address: string;
  items: ActivityItem[];
  next_before?: string;
}
